From bcf54dd3d99cfc1f3409f23d6f027399d228dc40 Mon Sep 17 00:00:00 2001
From: ysiyer <yegnesh.s.iyer@intel.com>
Date: Wed, 13 Apr 2016 16:29:36 +0530
Subject: [PATCH 1/2] XPWR CHARGER: Correcting the CV target voltage setting

Issue description: The driver was incorrectly checking for the
cv values. For example if the max voltage is 4.2V, then driver would
set the target voltage as 4.35V and for max voltage < 4.2V, the
driver was setting the target voltage to 4200mV. As a result, the
battery was never hitting "FULL" status as the charger would never
terminate charging based on ocv.

Solution: Modified the check from less than to less than equal to.
So for values <=4.2V, the target voltage would be 4.2V.

Change-Id: I44c84121cebc859c28f1b04bc4d27ab378989d98
Signed-off-by: ysiyer <yegnesh.s.iyer@intel.com>
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-22565
Reviewed-on: https://android.intel.com:443/494106
---
 drivers/power/dc_xpwr_charger.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/power/dc_xpwr_charger.c b/drivers/power/dc_xpwr_charger.c
index 80e52ff..fc13899 100644
--- a/drivers/power/dc_xpwr_charger.c
+++ b/drivers/power/dc_xpwr_charger.c
@@ -337,11 +337,11 @@ static inline int pmic_chrg_set_cv(struct pmic_chrg_info *info, int cv)
 	if (ret < 0)
 		goto set_cv_fail;
 
-	if (cv < CV_4100)
+	if (cv <= CV_4100)
 		reg_val = CHRG_CCCV_CV_4100MV;
-	else if (cv < CV_4150)
+	else if (cv <= CV_4150)
 		reg_val = CHRG_CCCV_CV_4150MV;
-	else if (cv < CV_4200)
+	else if (cv <= CV_4200)
 		reg_val = CHRG_CCCV_CV_4200MV;
 	else
 		reg_val = CHRG_CCCV_CV_4350MV;
-- 
1.9.1

